packages/svelte/src/compiler/phases/3-transform/client/visitors/Program.js

```diff
@@ -42,11 +42,11 @@ export function Program(_, context) {
 
 	for (const [name, binding] of context.state.scope.declarations) {
 		if (binding.kind === 'store_sub') {
-			const store = /** @type {Expression} */ (context.visit(b.id(name.slice(1))));
+			const get_store = () => /** @type {Expression} */ (context.visit(b.id(name.slice(1))));
```

Comment:
This is now done everytime this is accessed, which feels wasteful. Also a comment would be good to explain why we need this

Reply:
I know, but we're talking nanoseconds. Making the reads lazy feels like a more future-proof approach than carefully managing the order of operations, should we ever hit a similar scenario (we wouldn't want to add a _fourth_ iteration of `context.state.scope.declarations`)

Reply:
I'm not against this approach, all I'm saying is the method should write to a variable and return that in future invocations

Reply:
like this you mean?

```suggestion
			let cached;
			const get_store = () => (cached ??= /** @type {Expression} */ (context.visit(b.id(name.slice(1)))));
```

Reply:
yeah exactly 👍